Fix:

error: implicitly declaring library function 'exit' with type 'void (int) __attribute__((noreturn))' [-Werror,-Wimplicit-function-declaration]
error: implicitly declaring library function 'strcmp' with type 'int (const char *, const char *)' [-Werror,-Wimplicit-function-declaration]
error: implicit declaration of function 'XmRegisterConverters' is invalid in C99 [-Werror,-Wimplicit-function-declaration]

exit and strcmp are OS functions; fix by including the right OS header.

XmRegisterConverters is an internal Motif function; fix by providing the right declaration.
--- configure.orig	2015-02-14 16:59:46.000000000 -0600
+++ configure	2021-03-17 01:59:02.000000000 -0500
@@ -4935,6 +4935,7 @@
 /* end confdefs.h.  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 
 #define LOG2EA 0.44269504088896340735992
@@ -8179,8 +8180,9 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
-#include <stdio.h>
 #include <netcdf.h>
+#include <stdlib.h>
+#include <string.h>
       int main(void) {
         char *vlib;
         vlib = nc_inq_libvers();
@@ -8345,6 +8347,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+#include <stdlib.h>
 #include <string.h>
 #include <zlib.h>
       int main(void) {
@@ -8431,6 +8434,7 @@
 /* end confdefs.h.  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <jpeglib.h>
       int main(void) {
         int vinc;
@@ -8515,6 +8519,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+#include <stdlib.h>
 #include <string.h>
 #include <png.h>
       int main(void) {
@@ -8603,6 +8608,8 @@
 /* end confdefs.h.  */
 
 #include <pdflib.h>
+#include <stdlib.h>
+#include <string.h>
       int main(void) {
         char *vinc;
         int vlibn, vincn;
@@ -9552,7 +9559,9 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+#include <stdlib.h>
 #include <Xm/XmAll.h>
+void XmRegisterConverters(void);
       int main(void) {
         int vlibn, vincn;
         vincn = XmVersion;
